src/test/compile-fail/dropck-object-cycle.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright 2014 The Rust Project Developers. See the COPYRIGHT
```

Comment:
I mean, in both cases the problem is `TraitObject<'a>+'static : 'b`. This is safe under the non-syntactical rules, but has the problem of potentially having a witness that `'a` is alive. Of course, under the current rules, a `TraitObject<'a>+'static` must also be a `TraitObject<'static>+'static`, and is therefore safe to destroy under ordinary circumstances.